cdap-app-fabric/src/main/java/io/cdap/cdap/internal/bootstrap/BootstrapService.java

```diff
@@ -82,6 +82,12 @@ public class BootstrapService extends AbstractIdleService {
   protected void startUp() {
     LOG.info("Starting {}", getClass().getSimpleName());
     config = bootstrapConfigProvider.getConfig();
+
+    // Do not start other services if no bootstrap steps were executed.
+    if (config.getSteps().isEmpty()) {
+      return;
```

Comment:
can we please add an info log here before return?

Reply:
Done.